compressor/base.py

```diff
@@ -371,6 +371,10 @@ def render_output(self, mode, context=None):
 
         if hasattr(self.context, 'flatten'):
             # Passing Contexts to Template.render is deprecated since Django 1.8.
+            # NOTE the next line avoids https://github.com/django-compressor/django-compressor/issues/706
+            # by ensuring any object in self.context.dicts is "pre-flattened" before attempting to flatten
+            # the overall context.
+            self.context.dicts = list(map(lambda d: d.flatten() if hasattr(d, 'flatten') else d, self.context.dicts))
```

Comment:
Could we write this

```
self.context.dicts = [
       d.flatten() if hasattr(d, "flatten") else d 
       for d in self.context.dicts
]
```

I find it more readable than using list, map, and lambda.

Reply:
I tried some variant more like that and it did not work, maybe I had some bug. Feel free to push up an alternative version so long as it works and improves on the previous version by not dropping items from the list that don't have a flatten :-) I'm not particularly enthused by this version but it had the positive attribute of working...